spring-data-jdbc/src/main/java/org/springframework/data/jdbc/core/JdbcAggregateTemplate.java

```diff
@@ -139,13 +138,14 @@ public void setEntityCallbacks(EntityCallbacks entityCallbacks) {
 
 		Assert.notNull(instance, "Aggregate instance must not be null!");
 
-		RelationalPersistentEntity<?> persistentEntity = context.getRequiredPersistentEntity(instance.getClass());
-
-		Function<T, AggregateChangeWithRoot<T>> changeCreator = persistentEntity.isNew(instance)
-				? entity -> createInsertChange(prepareVersionForInsert(entity))
-				: entity -> createUpdateChange(prepareVersionForUpdate(entity));
+		return StreamSupport.stream(saveAll(singletonList(instance)).spliterator(), false).findFirst()
```

Comment:
Stream usage is always expensive and we should avoid it where possible. Ideally, we can have a `save` implementation that follows an optimized path for single-instance saves as batch saving isn't the predominant use case.